src/wallet/wallet.cpp

```diff
@@ -2602,94 +2602,111 @@ std::shared_ptr<CWallet> CWallet::Create(interfaces::Chain* chain, const std::st
+            if (min_tx_fee.value() == 0) {
+                error = AmountErrMsg("mintxfee", gArgs.GetArg("-mintxfee", ""));
+                return nullptr;
+            }
+
+            if (min_tx_fee.value() > HIGH_TX_FEE_PER_KB) {
             warnings.push_back(AmountHighWarn("-mintxfee") + Untranslated(" ") +
                                _("This is the minimum transaction fee you pay on every transaction."));
+            }
+
+            walletInstance->m_min_fee = CFeeRate(min_tx_fee.value());
+        } else {
+            error = AmountErrMsg("mintxfee", gArgs.GetArg("-mintxfee", ""));
+            return nullptr;
         }
-        walletInstance->m_min_fee = CFeeRate(n);
     }
 
     if (gArgs.IsArgSet("-maxapsfee")) {
         const std::string max_aps_fee{gArgs.GetArg("-maxapsfee", "")};
-        CAmount n = 0;
         if (max_aps_fee == "-1") {
-            n = -1;
-        } else if (!ParseMoney(max_aps_fee, n)) {
+            walletInstance->m_max_aps_fee = -1;
+        } else if (std::optional<CAmount> max_fee = ParseMoney(max_aps_fee)) {
+            if (max_fee.value() > HIGH_APS_FEE) {
+                warnings.push_back(AmountHighWarn("-maxapsfee") + Untranslated(" ") +
+                                  _("This is the maximum transaction fee you pay (in addition to the normal fee) to prioritize partial spend avoidance over regular coin selection."));
+            }
+            walletInstance->m_max_aps_fee = max_fee.value();
+        } else {
             error = AmountErrMsg("maxapsfee", max_aps_fee);
             return nullptr;
         }
-        if (n > HIGH_APS_FEE) {
-            warnings.push_back(AmountHighWarn("-maxapsfee") + Untranslated(" ") +
-                              _("This is the maximum transaction fee you pay (in addition to the normal fee) to prioritize partial spend avoidance over regular coin selection."));
-        }
-        walletInstance->m_max_aps_fee = n;
     }
 
     if (gArgs.IsArgSet("-fallbackfee")) {
-        CAmount nFeePerK = 0;
-        if (!ParseMoney(gArgs.GetArg("-fallbackfee", ""), nFeePerK)) {
-            error = strprintf(_("Invalid amount for -fallbackfee=<amount>: '%s'"), gArgs.GetArg("-fallbackfee", ""));
-            return nullptr;
-        }
-        if (nFeePerK > HIGH_TX_FEE_PER_KB) {
+        if (std::optional<CAmount> fallback_fee = ParseMoney(gArgs.GetArg("-fallbackfee", ""))) {
+
+            if (fallback_fee.value() > HIGH_TX_FEE_PER_KB) {
             warnings.push_back(AmountHighWarn("-fallbackfee") + Untranslated(" ") +
                                _("This is the transaction fee you may pay when fee estimates are not available."));
+            }
+
+            walletInstance->m_fallback_fee = CFeeRate(fallback_fee.value());
+
+            // Disable fallback fee in case value was set to 0, enable if non-null value
+            walletInstance->m_allow_fallback_fee = walletInstance->m_fallback_fee.GetFeePerK() != 0;
```

Comment:
Why is this allowed to move inside the `if (gArgs.IsArgSet("-fallbackfee")) {` code block? Prior to this change, `walletInstance->m_fallback_fee` would be set to the default (0), and `walletInstance->m_allow_fallback_fee` would therefore be flipped from its initialized value (true) to false.

Reply:
Think this was just a mistake.